modules/ximgproc/tutorials/disparity_filtering.markdown

```diff
@@ -0,0 +1,177 @@
+the disparity values from high- to low-confidence regions like half-occlusions. Recent advances in
+edge-aware filtering have enabled performing such post-filtering under the constraints of real-time
+processing on CPU.
+
+In this tutorial you will learn how to use the disparity map post-filtering to improve the results
+of StereoBM and StereoSGBM algorithms.
+
+Source Stereoscopic Image
+-------------------------
+
+![Left view](images/ambush_5_left.png)
+![Right view](images/ambush_5_right.png)
+
+Source Code
+-----------
+
+@code{.cpp}
+#include "opencv2/calib3d.hpp"
+#include "opencv2/ximgproc.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/core/utility.hpp"
+#include "opencv2/imgproc.hpp"
+
+using namespace cv;
+using namespace cv::ximgproc;
+using namespace std;
+
+const String keys =
+    "{help h usage ? || print this message}"
+    "{l || left view of the stereopair}"
+    "{r || right view of the stereopair}"
+    "{o || output disparity map}"
+    ;
+
+int main( int argc, const char** argv )
+{
+    CommandLineParser parser(argc,argv,keys);
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    String left_im  = parser.get<String>("l");
+    String right_im = parser.get<String>("r");
+    String dst_im   = parser.get<String>("o");
+
+    if ( !parser.check() )
+    {
+        parser.printErrors();
+        return -1;
+    }
+
+    Mat left  = imread(left_im);
+    Mat right = imread(right_im);
+    if ( left.empty() || right.empty() )
+    {
+        printf("Cannot read the source stereopair");
+        return -1;
+    }
+
+    Mat left_downscaled, right_downscaled;
+    resize(left, left_downscaled, Size(), 0.5, 0.5);
+    cvtColor(left_downscaled,  left_downscaled,  COLOR_BGR2GRAY);
+    resize(right, right_downscaled, Size(), 0.5, 0.5);
+    cvtColor(right_downscaled, right_downscaled, COLOR_BGR2GRAY);
+
+    Ptr<StereoBM> left_matcher = StereoBM::create(64,7);
+    Ptr<StereoMatcher> right_matcher;
+    Ptr<DisparityWLSFilter> wls_filter = createDisparityWLSFilter(left_matcher,right_matcher);
+
+    Mat left_disparity_map, right_disparity_map, res_disparity_map;
+    left_matcher ->compute(left_downscaled,  right_downscaled, left_disparity_map);
+    right_matcher->compute(right_downscaled, left_downscaled , right_disparity_map);
+    wls_filter->filter(left_disparity_map, left, res_disparity_map, right_disparity_map);
+
+    Mat disparity_map_visualization;
+    getDisparityVis(res_disparity_map,disparity_map_visualization,2.0);
+    if ( dst_im == "" )
+    {
+        namedWindow("filtered disparity map", WINDOW_AUTOSIZE);
+        imshow("filtered disparity map", disparity_map_visualization);
+        waitKey();
+    }
+    else
+        imwrite(dst_im, disparity_map_visualization);
+
+    return 0;
+}
+@endcode
+
+Explanation
+-----------
+-#  **Load left and right views**
+    @code{.cpp}
```

Comment:
Let's use snippets from original `disparity_filtering.cpp` example:
- First mark blocks in the example:
```.cpp
//! [some_block]
...
//! [some_block]
```
- Then use this block in the tutorial:
```.cpp
@snippet ximgproc/samples/disparity_filtering.cpp some_block
```